### ISSUE TITLE
Support upper/lower triangular masks with longer stride lengths

### ISSUE THREAD

**Issue by cathywu** [cathywu/TimeAlloc#45]
It's not that important that the contiguous scheduling is off by a half hour or a few hours. What's likely important is the number of days it takes to complete a task, so my stride can be `24*SLOTS_PER_HOUR` instead of 1.

Let's see how this affects optimization performance.

**Comment by cathywu** [comment]
Woohoo! Solve time 3.3 minutes (+1.7 min setup time) instead of 37.7 minutes (see https://github.com/cathywu/TimeAlloc/issues/34#issuecomment-424820901). Initial solution
```
Solve time 317.2721588611603
```

![image](https://user-images.githubusercontent.com/329117/46107924-90e2c700-c191-11e8-9b5f-10f3abef11ee.png)

Pretty good solution was found in 2 sec, and final solution wasn't improved after 22 sec.

```
Cbc0038I No solution found this major pass
Cbc0038I Before mini branch and bound, 1076 integers at bound fixed and 0 continuous
Cbc0038I Full problem 2979 rows 1465 columns, reduced to 616 rows 379 columns
Cbc0038I Mini branch and bound did not improve solution (2.86 seconds)
Cbc0038I After 2.86 seconds - Feasibility pump exiting with objective of -88.4012 - took 2.02 seconds
Cbc0012I Integer solution of -88.401245 found by feasibility pump after 0 iterations and 0 nodes (2.86 seconds)
Cbc0038I Full problem 2979 rows 1465 columns, reduced to 356 rows 205 columns
Cbc0031I 160 added rows had average density of 131.925
Cbc0013I At root node, 160 cuts changed objective from -88.412552 to -88.406273 in 10 passes
Cbc0014I Cut generator 0 (Probing) - 813 row cuts average 2.3 elements, 0 column cuts (54 active)  in 0.213 seconds - new frequency is 1
Cbc0014I Cut generator 1 (Gomory) - 365 row cuts average 366.2 elements, 0 column cuts (0 active)  in 0.473 seconds - new frequency is 1
Cbc0014I Cut generator 2 (Knapsack) - 0 row cuts average 0.0 elements, 0 column cuts (0 active)  in 0.052 seconds - new frequency is -100
Cbc0014I Cut generator 3 (Clique) - 0 row cuts average 0.0 elements, 0 column cuts (0 active)  in 0.002 seconds - new frequency is -100
Cbc0014I Cut generator 4 (MixedIntegerRounding2) - 0 row cuts average 0.0 elements, 0 column cuts (0 active)  in 0.043 seconds - new frequency is -100
Cbc0014I Cut generator 5 (FlowCover) - 0 row cuts average 0.0 elements, 0 column cuts (0 active)  in 0.015 seconds - new frequency is -100
Cbc0014I Cut generator 6 (TwoMirCuts) - 377 row cuts average 121.2 elements, 0 column cuts (0 active)  in 0.247 seconds - new frequency is -100
Cbc0010I After 0 nodes, 1 on tree, -88.401245 best solution, best possible -88.406273 (7.47 seconds)
Cbc0038I Full problem 2979 rows 1465 columns, reduced to 310 rows 157 columns
Cbc0012I Integer solution of -88.402327 found by RINS after 11927 iterations and 41 nodes (17.76 seconds)
Cbc0038I Full problem 2979 rows 1465 columns, reduced to 192 rows 104 columns
Cbc0012I Integer solution of -88.403139 found by RINS after 14873 iterations and 91 nodes (19.52 seconds)
Cbc0010I After 100 nodes, 43 on tree, -88.403139 best solution, best possible -88.406273 (19.79 seconds)
Cbc0038I Full problem 2979 rows 1465 columns, reduced to 178 rows 85 columns
Cbc0012I Integer solution of -88.40368 found by RINS after 26712 iterations and 200 nodes (22.99 seconds)
```
```
Objective value:                -88.40367965
Lower bound:                    -88.406
```

**Comment by cathywu** [comment]
There seems to be an issue, when the O'reilly event is forced to be on Sun/Mon, the `CTu_total` values remains the same...
```
    CTu_total : Size=1, Index=None
        Key  : Lower : Value      : Upper : Fixed : Stale : Domain
        None :  None : 0.20914502 :  None : False : False :  Reals
    CTl_total : Size=1, Index=None
        Key  : Lower : Value     : Upper : Fixed : Stale : Domain
        None :  None : 1.1926407 :  None : False : False :  Reals
```
![image](https://user-images.githubusercontent.com/329117/46108679-7a3d6f80-c193-11e8-80a0-587daa175121.png)

**Comment by cathywu** [comment]
Removed rows of tril and triu that was all ones:
```
    CTu_total : Size=1, Index=None
        Key  : Lower : Value      : Upper : Fixed : Stale : Domain
        None :  None : 0.20833333 :  None : False : False :  Reals
    CTl_total : Size=1, Index=None
        Key  : Lower : Value     : Upper : Fixed : Stale : Domain
        None :  None : 1.1943473 :  None : False : False :  Reals
```

![image](https://user-images.githubusercontent.com/329117/46109577-e0c38d00-c195-11e8-9af3-d3103ad0b84d.png)

**Comment by cathywu** [comment]
Fixed issue with denominators:
<img width="635" alt="image" src="https://user-images.githubusercontent.com/329117/46110433-76f8b280-c198-11e8-9b73-281b1b246969.png">

```
Objective value:                -86.46066434
Lower bound:                    -87.472
```
Note: Something unscheduled.
```
    A_total : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  86.0 :  None : False : False :  Reals
```
More reasonable:
```
    CTu_total : Size=1, Index=None
        Key  : Lower : Value      : Upper : Fixed : Stale : Domain
        None :  None : 0.22960373 :  None : False : False :  Reals
    CTl_total : Size=1, Index=None
        Key  : Lower : Value      : Upper : Fixed : Stale : Domain
        None :  None : 0.23106061 :  None : False : False :  Reals
```
Individual CTl, CTu values are all between 5 and 6 (with slack=5).

**Comment by cathywu** [comment]
Resolved by #47
